### PR TITLE
Fixes a crash when UIColor returns a color component with negative value

### PR DESCRIPTION
**Pull request by puelocesar**
I was getting this crash on my iPhone 7 test device because UIColor was returning a color component with a negative value (-0.005)

### CODE REVIEW

File: Sources/DynamicColor.swift [hunk 1]
```diff
@@ -110,6 +110,10 @@ public extension DynamicColor {
    */
   public final func toHex() -> UInt32 {
     func roundToHex(_ x: CGFloat) -> UInt32 {
+      if x < 0 {
```
Comment (yannickl) [inline Sources/DynamicColor.swift]: can you use something like that instead?
```swift
guard x > 0 else { return 0 }
```

It'll be more elegant.

**Comment by yannickl** [comment]
Wow, this is the first time I see that. In which case does it happen?

**Comment by puelocesar** [comment]
Yes, it's a very strange case. We are using some hue/saturation calculations to create UIColors, but I can't reproduce it anymore, it happened only once.

But, despite what Apple documentation says, it is possible to have an UIColor with negative color component

**Comment by yannickl** [comment]
Thank you for the fix! 👍

**Pull request merged**